File: src/main/java/handlers/EventBEnricher.java
package handlers;

import org.springframework.stereotype.Component;

/**
 * @author Evgeny Borisov
 */
@Component
public class EventBEnricher<T extends Event> implements EventEnricher<EventB> {
    @Override
    public Event enrichCurrent(EventB event) {
        System.out.println("handling BBB");
        System.out.println(event.getBbb());
        return event;
    }
}

File: src/main/scala/relfections/EventBEnricher.scala
package relfections

import org.springframework.stereotype.Component

/**
  * @author Evgeny Borisov
  */
@Component
class EventBEnricher extends EventEnricher[EventB] {


  override def myEvents(): Set[Class[_ <: Event]] = {
    Set(classOf[EventB])
  }

  override def enrich(event: EventB): EventB = {
    event
  }
}
